scripts/release-notes.py

```diff
@@ -88,17 +108,70 @@ def main(token, repo, num_commits, exclude_dependabot):
         if not pulls:
             short_sha = sha[:7]
             commit_url = commit['html_url']
-            print(f'* {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))')
+            # Check if the commit has changelog label
+            commit_labels = get_pull_request_labels(token, args.repo, pull_id)
+            changelog_labels = [label for label in commit_labels if label.startswith('changelog:')]
+
+            category = 'Other'
+            if changelog_labels:
+                for cat in categories:
+                    if changelog_labels[0].startswith(cat['label']):
+                        category = cat['title']
+                        break
+            
+            result = f'* {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))'
+            if category == 'Other':
+                other_results.append(result)
+            else:
+                category_results[category].append(result)
             continue
 
         pull = pulls[0]
         pull_id = pull['number']
         pull_url = pull['html_url']
         msg = msg.replace(f'(#{pull_id})', '').strip()
-        print(f'* {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))')
 
-    if skipped_dependabot:
-        print(f"\n(Skipped {skipped_dependabot} dependabot commit{'' if skipped_dependabot == 1 else 's'})")
+        # Check if the pull request has changelog label
+        pull_labels = get_pull_request_labels(token, args.repo, pull_id)
+        changelog_labels = [label for label in pull_labels if label.startswith('changelog:')]
+
+        category = 'Other'
+        if changelog_labels:
+            for cat in categories:
+                if changelog_labels[0].startswith(cat['label']):
+                    category = cat['title']
+                    break
+            
+        result = f'* {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))'
+        if category == 'Other':
+            other_results.append(result)
+        else:
+            category_results[category].append(result)
+
+    # Print categorized pull requests
+    for category, results in category_results.items():
+        if results and category:
+            print(f'{category}:')
+            for result in results:
+                print(result)
+            print()
+
+    # Print pull requests in the 'Other' category
+    if other_results:
+        print('#### Other:')
+        for result in other_results:
+            print(result)
+
+        if skipped_dependabot:
```

Comment:
should this be indented left?

Reply:
No. Shouldn't it stay inside the main function?